crates/handlers/src/rate_limit.rs

```diff
@@ -199,6 +237,66 @@ impl Limiter {
 
         Ok(())
     }
+
+    /// Check if an email can be sent to the address for an email
+    /// authentication session
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the operation is rate limited.
+    pub fn check_email_authentication_email(
+        &self,
+        requester: RequesterFingerprint,
+        email: &str,
+    ) -> Result<(), EmailAuthenticationLimitedError> {
+        self.inner
+            .email_authentication_per_requester
+            .check_key(&requester)
+            .map_err(|_| EmailAuthenticationLimitedError::Requester(requester))?;
+
+        // Convert to lowercase to prevent bypassing the limit by enumerating different
+        // case variations.
+        // A case-folding transformation may be more proper.
```

Comment:
maybe so, but this is likely good enough. I don't think non-ASCII e-mail localparts are common at all? I have never seen one in the real-world.

To be useful, we'd have to assume the recipient server is also doing casefolding transformations. I am sceptical.

All in all, happy with just lowercasing